app/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -116,49 +100,55 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
       is AdHocDataCollectionButtonData -> cardUiData.job.tasks.values.isNotEmpty()
     }
 
-  @Composable
-  private fun ShowDataSharingTermsDialog(
+  private fun showDataSharingTermsDialog(
     cardUiData: DataCollectionEntryPointData,
     dataSharingTerms: DataSharingTerms,
   ) {
-    DataSharingTermsDialog(dataSharingTerms) {
-      val job = lifecycleScope.launch { mapContainerViewModel.updateDataSharingConsent(true) }
-      job.cancel()
-      navigateToDataCollectionFragment(cardUiData)
+    renderComposableDialog {
+      DataSharingTermsDialog(dataSharingTerms) {
+        mapContainerViewModel.grantDataSharingConsent()
+        navigateToDataCollectionFragment(cardUiData)
+      }
     }
   }
 
   /** Invoked when user clicks on the map cards to collect data. */
-  private fun onCollectData(
-    canUserSubmitData: Boolean,
-    hasTasks: Boolean,
-    hasDataSharingTerms: DataSharingTerms?,
-    cardUiData: DataCollectionEntryPointData,
-  ) {
+  private fun onCollectData(canUserSubmitData: Boolean, cardUiData: DataCollectionEntryPointData) {
     if (!canUserSubmitData) {
       // Skip data collection screen if the user can't submit any data
       // TODO: Revisit UX for displaying view only mode
       // Issue URL: https://github.com/google/ground-android/issues/1667
       ephemeralPopups.ErrorPopup().show(getString(R.string.collect_data_viewer_error))
       return
     }
-    if (!hasTasks) {
+    if (!hasValidTasks(cardUiData)) {
       // NOTE(#2539): The DataCollectionFragment will crash if there are no tasks.
       ephemeralPopups.ErrorPopup().show(getString(R.string.no_tasks_error))
       return
     }
-    if (hasDataSharingTerms != null) {
-      if (
-        hasDataSharingTerms.type == DataSharingTerms.Type.CUSTOM &&
-          hasDataSharingTerms.customText.isBlank()
-      ) {
-        ephemeralPopups.ErrorPopup().show(getString(R.string.invalid_data_sharing_terms))
-        return
+
+    mapContainerViewModel
+      .getDataSharingTerms()
+      .onSuccess { terms ->
+        if (terms == null) {
+          // Data sharing terms already accepted or missing.
+          navigateToDataCollectionFragment(cardUiData)
```

Comment:
```suggestion
          // Data sharing terms already accepted.
          navigateToDataCollectionFragment(cardUiData)
```

right?

Reply:
We don't show the dialog if the survey hasn't configured a data sharing terms. This PR doesn't change the existing behavior.